### ISSUE TITLE
Imagick::morphology ignores channel

### ISSUE THREAD

**Issue by Kallys** [Imagick/imagick#379]
Hey!

I suspect `Imagick::morphology` method is ignoring the `$channel` argument.
I don't know which macro you are using but this line sounds suspect, no?
https://github.com/Imagick/imagick/blob/06116aa24b76edaf6b1693198f79e6c295eda8a9/imagick_class.c#L12642

I can provide example if needed, but basically:
```shell
$ convert in.tiff -channel A -morphology Erode diamond:1 +channel out.tiff
```
and
```php
$imagick->morphology(Imagick::MORPHOLOGY_ERODE, 1,  ImagickKernel::fromBuiltin(Imagick::KERNEL_DIAMOND, "1"), Imagick::CHANNEL_ALPHA);
```
are not producing same output (input image is in CMYK colorspace btw).

Thanks!

Version tested:
* imagick module version => 3.4.4
* Imagick compiled with ImageMagick version => ImageMagick 6.9.10-23 Q16 x86_64 20190101 https://imagemagick.org

**Comment by Danack** [comment]
That is pretty dodgy looking...

**Comment by Danack** [comment]
Was about to fix it, when I noticed that something else is pretty dodgy also: https://github.com/Imagick/imagick/commit/27dd723d6b7a7b43a384db8a358a10b31c013f95

So rather than being a trivial fix, I need to do some sanity checking on why that define is like that, and if changing it is appropriate.

**Comment by Danack** [comment]
The default channel stuff might be to do with ImageMagick using UndefinedChannel as all channels.
```
if (channel_mask != UndefinedChannel)
        (void) SetPixelChannelMask(image,channel_mask);
```
        
I need to write some test code to see what the different actually is across ImageMagick 6 + 7, to find the actual behaviour.

**Comment by Danack** [comment]
Ugh, I have no idea why the current default channel works at all. But as no-one has complained about it, I'll leave it for now.

I've opened an issue to change it at an appriopriate time. The channel not being set is fixed in 0348046050a8f9f027b3d0880844d56dbf1f7774

**Comment by Danack** [comment]
> Ugh, I have no idea why the current default channel works at all.

btw I'd forgotten that to avoid a huge BC break from upstream, some magic was used to check if the user had set the channel or not.

I [added a note](https://github.com/Imagick/imagick/blob/86d95fed787d60db110ccfeb2d89ec53f1d1f5f7/php_imagick_defs.h#L1245-L1252) to avoid having to remember in future.
